### DISCUSSION TITLE
Before spec running for every row in csv with -DinParallel=true

### DISCUSSION THREAD

**Discussion by bansal63** [getgauge/gauge#1843] [General]
I've a spec which has 

    table: data/csv/a.csv
    Tags: tag1
Where csv file has 49 cases and there is only 1 scenario in spec

Before spec in a StemImplementation looks like below

    	@BeforeSpec(tags = {"tag1","tag2"},tagAggregation = Operator.OR)
	    public static void preSpec() {
        }

When i run spec in parallel, it is running before spec 49 times.
It should be run as many thread i've.

gauge -v

    Gauge version: 1.1.1 Plugins
    html-report (4.0.12) 
    java (0.7.3) 
    json-report (0.3.3) 
    xml-report (0.2.3)

**Comment by sriv** [comment]
This is expected. The spec is run once per data row, and in parallel, i.e. each row creates a copy of the spec with the data specified.

Hence Gauge would run `beforespec` for every data row. The number of threads is irrlevant here, since it can change based on the machine or the arguments passed. And it must do this to maintain isolation between the parallely run specs.
